eogrow/pipelines/split_grid.py

```diff
@@ -0,0 +1,184 @@
+"""Implements a pipeline that creates a finer grid and splits EOPatches accordingly."""
+import logging
+from collections import defaultdict
+from typing import Dict, List, Literal, Tuple, Union
+
+import fs
+import geopandas as gpd
+from pydantic import Field
+
+from eolearn.core import EONode, EOWorkflow, FeatureType, LoadTask, OverwritePermission, SaveTask
+from sentinelhub import BBox
+
+from ..core.area.batch import BatchAreaManager
+from ..core.area.utm import UtmZoneAreaManager
+from ..core.pipeline import Pipeline
+from ..tasks.spatial import SpatialSliceTask
+from ..utils.fs import LocalFile
+from ..utils.grid import split_bbox
+from ..utils.types import Feature, FeatureSpec
+
+LOGGER = logging.getLogger(__name__)
+
+NamedBBox = Tuple[str, BBox]
+
+
+class SplitGridPipeline(Pipeline):
+    """Pipeline that creates a finer grid and splits EOPatches accordingly.
+
+    The new grid is output as a geopackage file, which can be used with a `CustomAreaManager`.
+    """
+
+    class Schema(Pipeline.Schema):
+        input_folder_key: str = Field(
+            description="A storage manager key pointing to the folder from where data will be loaded."
+        )
+        eopatch_output_folder_key: str = Field(
+            description="A storage manager key pointing to the folder to where data will be saved."
+        )
+        grid_output_folder_key: str = Field(
+            description="A storage manager key of where to save the resulting split grid."
+        )
+        subsplit_grid_filename: str = Field(
+            description="Filename of new grid, which can be used in `CustomAreaManager`.", regex=r"^.+\.gpkg$"
+        )
+        features: List[Feature] = Field(description="Which features will be loaded and joined.")
+        raise_misaligned: bool = Field(
+            True,
+            description=(
+                "If True this will raise an error if splitting or joining any spatial raster EOPatch feature would "
+                "cause a misalignment. If False, misalignment issues will be ignored."
+            ),
+        )
+
+        split_x: int = Field(2, description="Number of splits done on each EOPatch along the x-axis.")
+        split_y: int = Field(2, description="Number of splits done on each EOPatch along the y-axis.")
+
+        buffer: Union[Literal["auto"], Tuple[float, float]] = Field(
+            "auto",
+            description=(
+                "How large is the original EOPatch buffer, provided as `(buffer_x, buffer_y)` in CRS units. Applies"
+                " same buffer to split EOPatches. The pipeline tries to obtain this information from area manager, but"
+                " not all area manager classes are supported."
```

Comment:
Replace first sentence with:  "Amount of buffer present on existing EOPatches, provided as `(buffer_x, buffer_y)` in CRS units."

>  The pipeline tries to obtain this information from area manager, but not all area manager classes are supported.

What if the new process saves the buffer as a column in the geopackage? would this provide any value for easier interpretation in such cases?

Reply:
Changed docstring

Since all EOPatches have the same buffer, it would be quite a bit of memory for little gain. Also remains difficult for `CustomAreaManager`, where patches can have distinct buffer values in the column, which is also odd. So not sure.